packages/babel-types/src/validators/isScope.js

```diff
@@ -18,5 +19,9 @@ export default function isScope(node: Object, parent: Object): boolean {
     return false;
   }
 
+  if (isPattern(node) && isFunction(parent)) {
```

Comment:
If a `Pattern` is a child of `Function`, it must be one of the `params` key because both `body` and `typeAnnotation` can not _be_ a `Pattern`.

Reply:
If you need to add a comment to explain your PR, then why don't add it directly in the code? :wink: